File: apps/ios-swiftui/StockAnalysisApp/ViewModels/StockRepository.swift
import Foundation

struct StockRepository {
    func loadStocks() -> [Stock] {
        MockStocks.all
    }

    func loadSectors() -> [HotSector] {
        MockStocks.sectors
    }

    func loadNews() -> [NewsEntry] {
        MockStocks.news
    }
}

File: apps/flutter-app/lib/view_models/stock_repository.dart
import '../mocks/mock_stocks.dart';
import '../models/stock.dart';

class StockRepository {
  const StockRepository();

  List<Stock> loadStocks() => mockStocks;

  Stock? findStock(String code) {
    for (final stock in mockStocks) {
      if (stock.code == code) return stock;
    }
    return null;
  }
}

